pkg/controller/tas/topology_ungater.go

```diff
@@ -447,17 +446,17 @@ func readRanksForLabels(
 	podIndexLabel string,
 	rjInfo *replicatedJobsInfo,
 ) (map[int]*corev1.Pod, error) {
-	result := make(map[int]*corev1.Pod, 0)
+	result := make(map[int]*corev1.Pod)
 	podSetSize := int(*psa.Count)
 	for _, pod := range pods {
-		podIndex, err := readIntFromLabel(pod, podIndexLabel)
+		podIndex, err := utilpod.ReadUIntFromLabel(pod, podIndexLabel)
 		if err != nil {
 			// the Pod has no rank information - ranks cannot be used
 			return nil, err
 		}
 		rank := *podIndex
 		if rjInfo != nil {
-			jobIndex, err := readIntFromLabel(pod, rjInfo.jobIndexLabel)
+			jobIndex, err := utilpod.ReadUIntFromLabel(pod, rjInfo.jobIndexLabel)
```

Comment:
Here we can use `ReadUIntFromLabelBelowBound(pod, rjInfo.jobIndexLabel, rjInfo.replicasCount)`

Reply:
Done